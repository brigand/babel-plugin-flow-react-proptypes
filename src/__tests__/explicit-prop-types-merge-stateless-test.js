const babel = require('babel-core');
const content = `
var React = require('react');

type FooProps = {
  a: number,
  b: number,
  c: string,
}

const Foo = (props: FooProps) => {
  return (
    <div>
      {props.a}
      {props.b}
      {props.c}
      {props.d}
    </div>
  );
};

Foo.propTypes = {
  b: PropTypes.string,
  d: PropTypes.string
};

export default Foo;
`;

it('explicit-prop-types-merge-stateless-test', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
