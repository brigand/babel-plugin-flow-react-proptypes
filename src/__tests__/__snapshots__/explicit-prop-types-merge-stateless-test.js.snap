// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`explicit-prop-types-merge-stateless-test 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var React = require('react');

var Foo = function Foo(props) {
  return React.createElement(
    'div',
    null,
    props.a,
    props.b,
    props.c,
    props.d
  );
};

Foo.propTypes = {
  a: _propTypes2.default.number.isRequired,
  b: _propTypes2.default.string,
  c: _propTypes2.default.string.isRequired,
  d: _propTypes2.default.string
};
exports.default = Foo;"
`;
